drivers/hmis/app/models/hmis/form/form_processor.rb

```diff
@@ -401,6 +426,25 @@ def related_records
     end.compact.uniq
   end
 
+  def destroy_related_records!
+    [
+      :health_and_dv,
+      :income_benefit,
+      :physical_disability,
+      :developmental_disability,
+      :chronic_health_condition,
+      :hiv_aids,
+      :mental_health_disorder,
+      :substance_use_disorder,
+      :exit,
+      :youth_education_status,
+      :employment_education,
+      :current_living_situation,
+      :ce_assessment,
+      :ce_event,
+    ].map { |assoc| send(assoc)&.destroy! }.compact
```

Comment:
is the return value significant? Looks like it would either all succeed or raise on failure
```suggestion
    ].each { |assoc| send(assoc)&.destroy! }
```
```suggestion
    ].map { |assoc| send(assoc)&.destroy! }.compact
```
```suggestion
    ].map { |assoc| send(assoc)&.destroy! }.compact
```

Reply:
No, updated